docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -0,0 +1,546 @@
+
+Unlike `quarkus-websockets`, the `quarkus-websockets-next` extension does **not** implement the Jakarta WebSocket specification.
+Instead, it introduces a modern API, prioritizing simplicity of use.
+Additionally, it's tailored to integrate with Quarkus' reactive architecture and networking layer seamlessly.
+
+The annotations utilized by the Quarkus WebSockets next extension differ from those in JSR 356 despite, sometimes, sharing the same name.
+The JSR annotations carry a semantic that the Quarkus WebSockets Next extension does not follow.
+
+== Use the WebSockets Next extension
+
+To use the `websockets-next` extension, you need to add the `io.quarkus.quarkus-websockets-next` extension to your project.
+In your `pom.xml` file, add:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-websockets-next</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-websockets-next")
+----
+
+
+== Configure the WebSocket server
+
+The WebSocket handling reuses the _main_ HTTP server.
+
+Thus, the configuration of the WebSocket server is done in the `quarkus.http.` configuration section.
+
+WebSocket paths configured within the application are concatenated with the root path defined by `quarkus.http.root` (which defaults to /).
+This concatenation ensures that WebSocket endpoints are appropriately positioned within the application's URL structure.
+
+Refer to the xref:http-reference.adoc[HTTP guide] for more details.
+
+== Declare WebSocket endpoints
+
+To declare web socket endpoints, you need to create a class annotated with `@io.quarkus.websockets.next.WebSocket` and define the path of the WebSocket endpoint:
+
+[source,java]
+----
+package org.acme.websockets;
+
+import io.quarkus.websockets.next.WebSocket;
+import jakarta.inject.Inject;
+
+@WebSocket(path = "/chat/{username}")
+public class ChatWebSocket {
+
+}
+----
+
+Thus, client can connect to this web socket endpoint using `ws://localhost:8080/chat/your-name`.
+If TLS is used, the URL is `wss://localhost:8443/chat/your-name`.
+
+=== Path parameters
+
+The path of the WebSocket endpoint can contain path parameters.
+The syntax is the same as for JAX-RS resources: `{parameterName}`.
+
+Access to the path parameter values is done through the `io.quarkus.websockets.next.WebSocketConnection` _session_ object:
+
+[source,java]
+----
+@Inject io.quarkus.websockets.next.WebSocketConnection session;
+// ...
+String value = session.pathParam("parameterName");
+----
+
+Path parameter values are always strings.
+If the path parameter is not present in the path, the `pathParam` method returns `null`.
+
+NOTE: Query parameters are not supported.
```

Comment:
However, you can access the query string of the handshake request with `session.handshakeRequest().query()`.